src/ansys/fluent/core/solver/flobject.py

```diff
@@ -1058,26 +1058,50 @@ def get_active_child_names(self):
         """Names of children that are currently active."""
         ret = []
         for child in self.child_names:
-            if getattr(self, child).is_active():
+            if getattr(self, child).is_active() and not self._is_deprecated(child):
                 ret.append(child)
         return ret
 
     def get_active_command_names(self):
         """Names of commands that are currently active."""
         ret = []
         for command in self.command_names:
-            if getattr(self, command).is_active():
+            if getattr(self, command).is_active() and not self._is_deprecated(command):
                 ret.append(command)
         return ret
 
     def get_active_query_names(self):
         """Names of queries that are currently active."""
         ret = []
         for query in self.query_names:
-            if getattr(self, query).is_active():
+            if getattr(self, query).is_active() and not self._is_deprecated(query):
                 ret.append(query)
         return ret
 
+    def _is_deprecated(self, child_name: str) -> bool:
+        """If the 'child_name' is deprecated in a specific Fluent version.'"""
+        deprecated_version = getattr(self, child_name).get_attr("deprecated-version")
+        if deprecated_version and FluentVersion(self.version) >= FluentVersion(
+            deprecated_version
+        ):
+            return True
+        return False
```

Comment:
Can just return the value of the predicate directly.

Reply:
yes, Thanks.

Reply:
Done.